src/CodeGen_LLVM.cpp

```diff
@@ -1071,12 +1071,24 @@ void CodeGen_LLVM::optimize_module() {
     }
 #endif
 
+    if (get_target().has_feature(Target::ASAN)) {
+        auto addAddressSanitizerPass = [](const PassManagerBuilder &builder, legacy::PassManagerBase &pm) {
+            // TODO: we we (also/instead) want createAddressSanitizerModulePass?
+            pm.add(createAddressSanitizerFunctionPass());
```

Comment:
AFAIU asan initialization is done in createAddressSanitizerModulePass, so yes.

Reply:
Is that yes (both) or yes (instead)?

Reply:
Yes both.